src/main/java/reposense/authorship/FileInfoExtractor.java

```diff
@@ -56,10 +57,15 @@
     public static List<FileInfo> extractFileInfos(RepoConfiguration config) {
         logger.info("Extracting relevant file infos " + config.getLocation() + "...");
 
+        List<FileInfo> fileInfos = new ArrayList<>();
+
         // checks out to the latest commit of the date range to ensure the FileInfo generated correspond to the
         // git blame file analyze output
-        GitChecker.checkoutToDate(config.getRepoRoot(), config.getBranch(), config.getUntilDate());
-        List<FileInfo> fileInfos = new ArrayList<>();
+        try {
+            GitChecker.checkoutToDate(config.getRepoRoot(), config.getBranch(), config.getUntilDate());
```

Comment:
Not sure if relevant to this PR, but perhaps we could shift the checking out to `untilDate` commit be done right after we download the repo, so we can just skip generating any report for that repo.

Reply:
Hmm, that would be an architecture tweak. Considering that this change is already beyond the scope of this PR, that should be done in separate PR.